pkg/cmd/delete_credential.go

```
@@ -97,19 +90,64 @@ func (d deleteCredentialCmd) runPrompt() CommandRunnerFunc {
 	}
 }
 
+func (d *deleteCredentialCmd) resolveInput(cmd *cobra.Command, context string) (inputConfig, error) {
+	if IsFlagInput(cmd) {
+		return d.resolveFlags(cmd)
+	} else {
+		return d.resolvePrompt(context)
+	}
+}
+
+func (d *deleteCredentialCmd) resolvePrompt(context string) (inputConfig, error) {
+	data, err := d.ReadCredentialsValueInEnv(d.CredentialsPath(), context)
+	if err != nil {
+		return inputConfig{}, err
+	}
+
+	if len(data) == 0 {
+		return inputConfig{}, errors.New("you have no defined credentials in this env")
+	}
+
+	providers := make([]string, len(data))
+	for _, c := range data {
+		providers = append(providers, c.Provider)
+	}
+
+	provider, err := d.List("Credentials: ", providers)
+	if err != nil {
+		return inputConfig{}, err
+	}
+
+	if b, err := d.Bool("Are you sure want to delete this credential?", []string{"yes", "no"}); err != nil {
+		return inputConfig{}, err
+	} else if !b {
+		return inputConfig{}, nil
+	}
+	return inputConfig{provider}, nil
+}
+
+func (d *deleteCredentialCmd) resolveFlags(cmd *cobra.Command) (inputConfig, error) {
+	provider, err := cmd.Flags().GetString(providerFlagName)
+	if err != nil {
+		return inputConfig{}, err
+	}
+	return inputConfig{provider}, nil
```

you should return an error if the provider is empty